src/session_pool/session.js

```diff
@@ -200,17 +202,95 @@ export class Session {
     putResponse(response) {
```

Comment:
Frankly, I would have no idea what this function does just from its name. Why not call it `setCookiesFromResponse` ? Such name would make it super clear. Or, the `setCookies` function could support working with `Response` object and then we wouldn't need this extra function at all.